frontend/coprs_frontend/coprs/views/coprs_ns/coprs_general.py

```diff
@@ -1100,6 +1101,16 @@ def render_copr_fork(copr, form, confirm=False):
 def copr_fork_post(copr):
     form = forms.CoprForkFormFactory.create_form_cls(copr=copr, user=flask.g.user, groups=flask.g.user.user_groups)()
     if form.validate_on_submit():
+        limit = app.config["FORK_PACKAGES_LIMIT"]
+        packages_count = PackagesLogic.get_all(copr.id).count()
```

Comment:
I'm thinking about #2523.  Do you think we could have also a configuration value like "FORK_PROJECT_DENYLIST" with a manually maintained list of projects?

Speaking of convenience, it would be nice to have a `copr.is_forkable()` method
wrapping this (ideally redis-cached).

Reply:
Perhaps it is not a bad idea to add `copr.forks_disallowed` item into the DB table.
We could easily auto-enable/disable that field while we are analyzing the storage.

Reply:
> Speaking of convenience, it would be nice to have a copr.is_forkable() method wrapping this

100% agree, I moved the logic to a separate function, but it is `ComplexLogic.raise_if_cant_fork` which is more convenient. 

> I'm thinking about https://github.com/fedora-copr/copr/issues/2523. Do you think we could have also a configuration value like "FORK_PROJECT_DENYLIST" with a manually maintained list of projects?
> Perhaps it is not a bad idea to add copr.forks_disallowed item into the DB table.
We could easily auto-enable/disable that field while we are analyzing the storage.

This is something that I am not that thrilled about. The `FORK_PROJECT_DENYLIST` option can be implemented in a few minutes, so I have no problem with this.

The rest sounds like overkill to me.

Personally, I would prefer counting the size of backend data that is to be forked, and failing the action if it is too much. But that is not so user-friendly until we have #1108 implemented.